source/RSRCFilter.cpp

```diff
@@ -0,0 +1,43 @@
+#include "RSRCFilter.h"
+
+#include <MimeType.h>
+#include <string.h>
+
+static const char* supported_types[] = {
+	"application/x-vnd.be-elfexecutable",
+	"application/octet-stream",
+	NULL
+};
+
+bool
+RSRCFilter::Filter(const entry_ref* ref, BNode* node, struct stat_beos* st,
+	const char* filetype)
+{
+	BEntry entry(ref, true);
+	if (entry.InitCheck() != B_OK)
+		return false;
+	if (entry.IsDirectory())
+		return true;
+
+	entry_ref entryRef;
+	if (entry.GetRef(&entryRef) != B_OK)
+		return false;
+
```

Comment:
If `ref` is a symlink, then `filetype` is useless, and you have to use the file type of `entryRef` instead.

Reply:
How should I check if `ref` is a symlink? And, `entryRef` is an `entry_ref`` object, how can I get a file type from that?

I've searched around, asking mentors is always a last resort :)

Reply:
Check [here](https://github.com/HaikuArchives/Filer/blob/master/sources/TypedRefFilter.cpp#L58).